### PR TITLE
[FIZZ-199] new shared library package used by GC extractor

### PR DESCRIPTION
**Pull request by bradbanister**

### CODE REVIEW

**Comment by bradbanister** [comment]
This is actually my FIZZ-200 spike work, but is ready to be carried forward for FIZZ-199.  See https://tracker.ed-fi.org/browse/FIZZ-200

**Comment by stephenfuqua** [comment]
Looks nice so far. Do not have time to pull down this code and try it out locally. Did you get VS Code to behave nicely?

**Comment by bradbanister** [comment]
> Looks nice so far. Do not have time to pull down this code and try it out locally. Did you get VS Code to behave nicely?

It ended up being one of those "reload VS Code and all is well" issues.  🤷‍♂️

**Review by lalrg — COMMENTED** [review]
It looks good as a spike, it matches the ac for the story but I don't know if having a single project for all the shared code is the best approach or if it would be better to have projects as components of the solution, for example a project for LMS-FileSystem, other for LMS-SyncData and so on.

@bradbanister should I approve it now? or you are going to work on FIZZ-199 in this same branch?

**Comment by bradbanister** [comment]
> It looks good as a spike, it matches the ac for the story but I don't know if having a single project for all the shared code is the best approach or if it would be better to have projects as components of the solution, for example a project for LMS-FileSystem, other for LMS-SyncData and so on.

I'm thinking we start with a single shared package.  We can split later if it becomes too big/starts lacking cohesion.

> 
> @bradbanister should I approve it now? or you are going to work on FIZZ-199 in this same branch?

No need to approve, I'm continuing work here.

**Comment by lalrg** [comment]
> I'm thinking we start with a single shared package. We can split later if it becomes too big/starts lacking cohesion.

Agreed ✔️

File: .teamcity/_Self/GoogleClassroom/Project.kt
Comment (stephenfuqua) [inline .teamcity/_Self/GoogleClassroom/Project.kt:24]: look at those extra spaces! 

😆 

I don't really care.

Comment (bradbanister) [reply .teamcity/_Self/GoogleClassroom/Project.kt]: Actually, this is a good catch. They were tabs inserted by Notepad++.  Serves me right for not using VS Code!

**Comment by bradbanister** [comment]
This is now ready for real review.  After pulling down, run a "poetry install" on both GC and Canvas extractors to get the link to the shared library.

File: src/canvas-extractor/canvas_extractor/__main__.py
Comment (stephenfuqua) [inline src/canvas-extractor/canvas_extractor/__main__.py:141]: Would there be any value to calculating the datetime once and passing the same file to each `write_xyz` statement?

File: src/canvas-extractor/canvas_extractor/__main__.py
Comment (stephenfuqua) [inline src/canvas-extractor/canvas_extractor/__main__.py:124]: What do you think about moving these log statements into the shared library? Is there any unwanted impact in terms of  the output logged to the screen/file?

**Review by stephenfuqua — APPROVED** [review]
Couple of questions to consider, but I'm not going to hold up the merge for them. @lalrg please also take a look.

@bradbanister @lalrg please merge whenever you're satisfied, without needing any more review from me.

ps. the build is now passing :-).

**Review by lalrg — COMMENTED** [review]
Everything is looking good, please add src/canvas-extractor/tests/api/test.db to gitignore and remove it from the PR and after that you can merge

File: src/canvas-extractor/canvas_extractor/__main__.py
Comment (lalrg) [reply src/canvas-extractor/canvas_extractor/__main__.py:124]: I don't have any strong opinion on favour or against this change.

**Pull request merged**
